mlxlink/modules/mlxlink_utils.cpp

```diff
@@ -1269,6 +1269,25 @@ double mw_to_dbm(double x)
     return 10 * log10(x);
 }
 
+string floatToStr(float num, int resolution)
+{
+    char fmt[32];
+
+    if (!(num - (int)num))
+    {
+        snprintf(fmt, sizeof(fmt), "%.0f", num);
+    }
+
+    else if (resolution < 0)
+    {
+        snprintf(fmt, sizeof(fmt), "%f", num);
+    }
+
+    snprintf(fmt, sizeof(fmt), "%.*f", resolution, num);
+
+    return string(fmt);
+}
+
```

Comment:
please clarify the logic of this function

As it is now, whatever value is written to the "fmt"-buffer in lines 1278 or 1286 will then be overwritten in the line 1286. So what is the point to have lines 1276-1284?

futhermore please use C++ static_cast rather than C-style casting (https://www.cprogramming.com/reference/typecasting/staticcast.html#:~:text=Static%20casts%20are%20prefered%20over,the%20type%20to%20convert%20to.)